src/data/biomes/chargestone-cave.ts

```diff
@@ -0,0 +1,146 @@
+    [TimeOfDay.NIGHT]: [],
+    [TimeOfDay.ALL]: [
+      SpeciesId.MAGNEMITE,
+      SpeciesId.NOSEPASS,
+      SpeciesId.PLUSLE,
+      SpeciesId.MINUN,
+      SpeciesId.ROGGENROLA,
+      SpeciesId.JOLTIK,
+      SpeciesId.GRUBBIN,
+      SpeciesId.ALOLA_DIGLETT,
+      SpeciesId.ALOLA_GEODUDE,
+    ],
+  },
+  [BiomePoolTier.UNCOMMON]: {
+    [TimeOfDay.DAWN]: [],
+    [TimeOfDay.DAY]: [],
+    [TimeOfDay.DUSK]: [],
+    [TimeOfDay.NIGHT]: [],
+    [TimeOfDay.ALL]: [SpeciesId.PORYGON, SpeciesId.SABLEYE, SpeciesId.KLINK, SpeciesId.ORBEETLE, SpeciesId.SOLOSIS],
+  },
+  [BiomePoolTier.RARE]: {
+    [TimeOfDay.DAWN]: [],
+    [TimeOfDay.DAY]: [],
+    [TimeOfDay.DUSK]: [],
+    [TimeOfDay.NIGHT]: [],
+    [TimeOfDay.ALL]: [
+      SpeciesId.ELEKID,
+      SpeciesId.DUNSPARCE,
+      SpeciesId.DRILBUR,
+      SpeciesId.FERROSEED,
+      SpeciesId.TYNAMO,
+      SpeciesId.CHESPIN,
+      SpeciesId.MINIOR,
+    ],
+  },
+  [BiomePoolTier.SUPER_RARE]: {
+    [TimeOfDay.DAWN]: [],
+    [TimeOfDay.DAY]: [],
+    [TimeOfDay.DUSK]: [],
+    [TimeOfDay.NIGHT]: [],
+    [TimeOfDay.ALL]: [SpeciesId.BELDUM, SpeciesId.ROTOM],
+  },
+  [BiomePoolTier.ULTRA_RARE]: {
+    [TimeOfDay.DAWN]: [],
+    [TimeOfDay.DAY]: [],
+    [TimeOfDay.DUSK]: [],
+    [TimeOfDay.NIGHT]: [],
+    [TimeOfDay.ALL]: [SpeciesId.SANDY_SHOCKS, SpeciesId.IRON_THORNS],
+  },
+  [BiomePoolTier.BOSS]: {
+    [TimeOfDay.DAWN]: [],
+    [TimeOfDay.DAY]: [],
+    [TimeOfDay.DUSK]: [],
+    [TimeOfDay.NIGHT]: [],
+    [TimeOfDay.ALL]: [
+      SpeciesId.MAGNEZONE,
+      SpeciesId.PROBOPASS,
+      SpeciesId.GIGALITH,
+      SpeciesId.GALVANTULA,
+      SpeciesId.VIKAVOLT,
+      SpeciesId.ALOLA_GOLEM,
+      SpeciesId.ALOLA_DUGTRIO,
+    ],
+  },
+  [BiomePoolTier.BOSS_RARE]: {
+    [TimeOfDay.DAWN]: [],
+    [TimeOfDay.DAY]: [],
+    [TimeOfDay.DUSK]: [],
+    [TimeOfDay.NIGHT]: [],
+    [TimeOfDay.ALL]: [
+      SpeciesId.ELECTIVIRE,
+      SpeciesId.DUDUNSPARCE,
+      SpeciesId.EXCADRILL,
+      SpeciesId.FERROTHORN,
+      SpeciesId.EELEKTROSS,
+      SpeciesId.ROTOM,
+    ],
+  },
+  [BiomePoolTier.BOSS_SUPER_RARE]: {
+    [TimeOfDay.DAWN]: [],
+    [TimeOfDay.DAY]: [],
+    [TimeOfDay.DUSK]: [],
+    [TimeOfDay.NIGHT]: [],
+    [TimeOfDay.ALL]: [SpeciesId.METAGROSS, SpeciesId.SANDY_SHOCKS, SpeciesId.IRON_THORNS],
+  },
+  [BiomePoolTier.BOSS_ULTRA_RARE]: {
+    [TimeOfDay.DAWN]: [],
+    [TimeOfDay.DAY]: [],
+    [TimeOfDay.DUSK]: [],
+    [TimeOfDay.NIGHT]: [],
+    [TimeOfDay.ALL]: [SpeciesId.ZEKROM],
+  },
+};
+
+const trainerPool = {
+  [BiomePoolTier.COMMON]: [TrainerType.SCIENTIST, TrainerType.HIKER],
+  [BiomePoolTier.UNCOMMON]: [TrainerType.ACE_TRAINER],
+  [BiomePoolTier.RARE]: [],
+  [BiomePoolTier.SUPER_RARE]: [],
+  [BiomePoolTier.ULTRA_RARE]: [],
+  [BiomePoolTier.BOSS]: [TrainerType.CLAY, TrainerType.ELESA],
+  [BiomePoolTier.BOSS_RARE]: [],
+  [BiomePoolTier.BOSS_SUPER_RARE]: [],
+  [BiomePoolTier.BOSS_ULTRA_RARE]: [],
+};
+
+const weatherPool = {
```

Comment:
Don't we have interfaces at the ready for those? So the compiler would catch it if something was missing?